ntp-proto/src/time_types.rs

```diff
@@ -182,7 +182,17 @@ impl NtpDuration {
 
     /// Interpret an exponent `k` as `2^k` seconds, expressed as an NtpDuration
     pub(crate) fn from_exponent(input: i8) -> Self {
-        Self::from_seconds(2.0f64.powi(input as i32))
+        Self {
+            duration: if input > 30 {
```

Comment:
I will again suggest writing this as a `match` expression, which seems to end up more readable.